### ISSUE TITLE
Wrong result with `_.pickBy` in 4.17.0

### ISSUE THREAD

**Issue by holm** [lodash/lodash#2808]
In 4.17.0, `_.pickBy` returns unexpected results:

```
const _ = require('lodash');

const obj = {
  'ab.c': 'foo',
  abc: 'bar'
};

const result = _.pickBy(obj, (value, key) => key.endsWith('c'));

console.log(result);
```

This will print out:

```
{ ab: { c: 'foo' }, abc: 'bar' }
```

It seems the change in https://github.com/lodash/lodash/commit/c6ff776147b56c80c69d7b44e81d0241cf225a14 now splits it by dot, which I suspect is not intended.

**Comment by jvah** [comment]
I have to say the idea to make `_.pick` to work with deep nesting feels quite stupid to me, shouldn't this have a completely new function? We are currently using lodash like this:

```
> _.pick({'foo.txt': 'foo', 'bar.txt': 'bar', 'baz.txt': 'baz'}, ['foo.txt', 'baz.txt'])
{ 'foo.txt': 'foo', 'baz.txt': 'baz' }
```

With lodash 4.17.0 the result is now:

```
> _.pick({'foo.txt': 'foo', 'bar.txt': 'bar', 'baz.txt': 'baz'}, ['foo.txt', 'baz.txt'])
{ foo: { txt: 'foo' }, baz: { txt: 'baz' } }
```

This seems to be completely wrong to me, because pick should NEVER modify the object structure in question.

**Comment by jdalton** [comment]
Thanks for the heads up @holm!

**Comment by jdalton** [comment]
@holm

I've addressed this issue with #2826 so you won't have to use the an array escape for your use case.
I'll have a v4.17.2 patch release out sometime today.
